### ISSUE TITLE
Consolidate values API for CNI (values.cni vs values.istio_cni)

### ISSUE THREAD

**Issue by dgn** [istio/istio#49290]
We currently have two almost-identical looking fields in the values that are concerned with CNI. The first is `values.cni` which if `enabled` is set to `true` will install CNI with the settings as they are configured in `values.cni`.

There's another field called `values.istio_cni` which only has two settings:

```
values:
  istio_cni:
    # if true, istiod will attempt to use an existing CNI installation
    enabled: true
    # has an effect on some annotations in the injection template
    chained: true
```

So the distinction between them is that `values.cni` is the actual CNI configuration and installation, so it works like other parts of the values API, but `istio_cni` actually just configures istiod and the injection template to use an existing CNI installation - the distinction is necessary for cases like remote istiod, where you might want to enable CNI usage, but not install it, or if you're managing the lifecycles of istiod and CNI separately and want to be able to enable CNI usage without having to reinstall.

But I think the naming is very confusing, so I would very much prefer if we could move `istio_cni` to `values.pilot.cni`, as the istiod(-remote) chart is really the only chart that looks at it. Another problem is in the actual values API: we use the same protobuf message for both `cni` and `istio_cni` even though `istio_cni` supports just a tiny subset of the fields in `cni`.

**Identified Tasks:**
- [x] Separate CNIConfig and CNIUsageConfig protobuf messages in `values_types.proto` (covered in https://github.com/istio/istio/pull/49319)
- [x] Move `istio_cni` to `pilot.cni` with a fallback to `istio_cni` for backwards-compatibility (see https://github.com/istio/istio/pull/49352)
- [x] Deprecate usage of `istio_cni` and warn users that it will be removed in the future (done as part of https://github.com/istio/istio/pull/49495)
- [x] Deprecate `chained` in CNIUsageConfig, use `provider` instead (https://github.com/istio/istio/pull/49495)
- [x] Remove `istio_cni` (minimum two releases after the other tasks, targeted for 1.24 right now)

**Comment by bleggett** [comment]
Even if we have a different top-level key, having both `pilot.cni.enabled` and `components.cni.enabled`, where they are entirely disjoint, is a chart smell.


In charts, `components.cni.enabled` means the same thing as `pilot.cni.enabled` outside of the pilot chart (that is, we use it to control templating logic, and as a side effect, with _some_ tooling, it also has the side effect of actually installing the component). it means the same thing inside of the pilot chart as well, but doesn't have a side effect. This is unnecessarily opaque to anyone not already familiar with how Istio works.


- I would rather rename `pilot.cni.enabled` to something that actually reflects what it does - ``pilot.inject_redirect_validation_init` or something,.
- `pilot.cni.chained` has nothing to do with CNI really, and everywhere it _is_ used, it is only used to inject (or not) a Multus-specific annotation AFAICT like `k8s.v1.cni.cncf.io/networks` - it should be generalized to `pilot.cni_networks_annotation`

**Comment by dgn** [comment]
I agree with generalizing `chained`, but: I didn't touch it because we're working on a solution for OpenShift that doesn't use multus, so this whole field would not be needed anymore. That doesn't mean we can't refactor, it just wasn't a priority for me.

As for `pilot.inject_redirect_validation_init` - I think that makes it even harder to understand for users, and it's also not entirely what it does. It doesn't just inject the validation init container, it also disables the iptables init container, rendering a proxy dysfunctional if you don't have CNI installed. Effectively, the switch enables usage of CNI, I think that's the important part from a user's perspective

**Comment by bleggett** [comment]
For the chart in which the flag is scoped, the flag controls nothing but init container injection. We don't actually do global flag scoping, but the flag only makes sense to users in a globally scoped context, so it's unavoidably and admittedly a bit awkward no matter how we slice it.

**Comment by dgn** [comment]
Just to keep latest state in this issue: we agreed that `pilot.cni.enabled` will stay as the name to enable usage of CNI. `pilot.cni.chained` will be renamed to `pilot.cni.provider` where `chained: false` equals `provider: multus`, and `chained: true` equals `provider: default`. The reasoning here was that the flag really only adds a multus-specific annotation in the injection template, it is not directly related to the CNI plugin being executed as chained or standalone.

**Comment by howardjohn** [comment]
anything left here?

**Comment by bleggett** [comment]
No, I think `istio_cni.xx` is gone now.

**Comment by bleggett** [comment]
@dgn feel free to reopen if there was something else here that still needs capturing, but this can be closed AFAIK.
